latest/tests/test_validation.py

```diff
@@ -1,73 +1,85 @@
-    with open(testfile) as f:
-        data = ''.join(line for line in f if not line.lstrip().startswith('//'))
-        jsondata = json.loads(data)
-        validator.validate(jsondata)
-        strict_validator.validate(jsondata)
-
-
-@pytest.mark.parametrize("testfile", valid_files, ids=ids(valid_files))
-def test_valid_files(testfile):
-    with open(testfile) as f:
-        json_file = json.load(f)
-        validator.validate(json_file)
-        with pytest.raises(ValidationError):
-            strict_validator.validate(json_file)
-
-
-@pytest.mark.parametrize("testfile", invalid_files, ids=ids(invalid_files))
-def test_invalid(testfile):
-    with open(testfile) as f:
-        json_file = json.load(f)
-        with pytest.raises(ValidationError):
-            validator.validate(json_file)
-        with pytest.raises(ValidationError):
-            strict_validator.validate(json_file)
-
-
-@pytest.mark.xfail
-@pytest.mark.parametrize(
-    "testfile", invalid_but_dont_fail_files,
-    ids=ids(invalid_but_dont_fail_files))
-def test_invalid_but_dontfail(testfile):
-    with open(testfile) as f:
-        json_file = json.load(f)
-        with pytest.raises(ValidationError):
-            validator.validate(json_file)
-        with pytest.raises(ValidationError):
-            strict_validator.validate(json_file)
+@dataclass
+class Suite:
+    schema:  dict
+    schema_store: dict
+    data: dict
+    valid: bool = True
+
+    def validate(self, validator) -> None:
+        if not self.valid:
+            with pytest.raises(ValidationError):
+                validator.validate(self.data)
+        else:
+            validator.validate(self.data)
+
+
+def pytest_generate_tests(metafunc):
+    """
+    Generates tests for the examples/ as well as tests/ subdirectories.
+
+    Examples:
+        These tests evalute all of the files under the examples/ directory
+        using the configuration in the provided config file in order detect
+        what should be run. It is assumed that all files are valid and complete
+        so that they can be wholly included into the specification. The
+        .config.json file in each directory defines which schema will be used.
+    
+    Validation:
+        These test consumes https://github.com/json-schema-org/JSON-Schema-Test-Suite#structure-of-a-test
+        styled JSON tests. Metadata in each test defines which schema is used
+        and whether or not the block is considered valid.
+    """
+    if "suite" in metafunc.fixturenames:
+        suites: List[Schema] = []
+        ids: List[str] = []
+
+        # Validation
+        for filename in glob.glob("tests/*.json"):
+            with open(filename) as o:
+                suite = json.load(o)
+            schema = suite["schema"]
+            with open(schema["id"]) as f:
+                schema = json.load(f)
+            for test in suite["tests"]:
+                ids.append("validate_" + str(test["formerly"]).split("/")[-1][0:-5])
```

Comment:
Possibly points to using the `id` key - see https://github.com/json-schema-org/JSON-Schema-Test-Suite/blob/master/test-schema.json#L61 ?

Reply:
Yes, definitely happy to move away from these for the proper suite ones. For examples, we'll still need to generate.